CMakeLists.txt

```
@@ -63,8 +64,21 @@ configure_file("${CMAKE_SOURCE_DIR}/cmake/templates/license.h.in" include/licens
 
 include(EthOptions)
 configure_project(TESTS)
+set(LATEST_Z3_VERSION "4.8.12")
+if (${STRICT_Z3_VERSION})
+    find_package(Z3 ${LATEST_Z3_VERSION} EXACT)
+    if (${USE_Z3} AND NOT ${Z3_FOUND})
+      message(
+        FATAL_ERROR
+        "SMTChecker tests require Z3 ${LATEST_Z3_VERSION} for all tests to pass.\n\
+Build with -DSTRICT_Z3_VERSION=OFF if you want to use an older version. \
+You can also use -DUSE_Z3=OFF to build without Z3. In both cases use --no-smt when running tests."
+            )
+    endif()
+else()
+    find_package(Z3 4.8.0 REQUIRED)
+endif()
```

Indents are a bit off. This file uses 2 spaces so this code should too:

```suggestion
if (${STRICT_Z3_VERSION})
  find_package(Z3 ${LATEST_Z3_VERSION} EXACT)
  if (${USE_Z3} AND NOT ${Z3_FOUND})
    message(
      FATAL_ERROR
        "SMTChecker tests require Z3 ${LATEST_Z3_VERSION} for all tests to pass.\n\
Build with -DSTRICT_Z3_VERSION=OFF if you want to use an older version. \
You can also use -DUSE_Z3=OFF to build without Z3. In both cases use --no-smt when running tests."
    )
  endif()
else()
  find_package(Z3 4.8.0 REQUIRED)
endif()
```
Though we should really reformat it with tabs to match our [style](https://github.com/ethereum/solidity/blob/develop/.editorconfig#L21-L23) at some point. 2 space indents are just awful for readability.